src/main/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommand.java

```diff
@@ -0,0 +1,101 @@
+package seedu.sellsavvy.logic.commands.ordercommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.sellsavvy.logic.Messages.MESSAGE_ORDERLIST_DOES_NOT_EXIST;
+
+import seedu.sellsavvy.commons.core.index.Index;
+import seedu.sellsavvy.commons.util.ToStringBuilder;
+import seedu.sellsavvy.logic.Messages;
+import seedu.sellsavvy.logic.commands.Command;
+import seedu.sellsavvy.logic.commands.CommandResult;
+import seedu.sellsavvy.logic.commands.exceptions.CommandException;
+import seedu.sellsavvy.model.Model;
+import seedu.sellsavvy.model.order.Count;
+import seedu.sellsavvy.model.order.Date;
+import seedu.sellsavvy.model.order.Item;
+import seedu.sellsavvy.model.order.Order;
+import seedu.sellsavvy.model.order.OrderList;
+import seedu.sellsavvy.model.order.Status;
+import seedu.sellsavvy.model.person.Person;
+
+/**
+ * Marks an order as completed.
+ */
+public class MarkOrderCommand extends Command {
+
+    public static final String COMMAND_WORD = "markOrder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Marks an order as completed."
+            + "Parameters: ORDER_INDEX (must be positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 2";
+
+    public static final String MESSAGE_MARK_ORDER_SUCCESS = "The order has been marked as completed: %1$s";
+
+    private final Index index;
+
+    /**
+     * Creates a MarkOrderCommand to mark the order under the
+     * specified index as completed.
+     *
+     * @param index of the order in the displayed order list to mark.
+     */
+    public MarkOrderCommand(Index index) {
+        requireNonNull(index);
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Person selectedPerson = model.getSelectedPerson().get();
```

Comment:
![image](https://github.com/user-attachments/assets/90c40b18-240c-41e9-b676-5dc3ae22dcad)
I added all these method for my part to follow law of Demeter. I think I will just edit for you later since my part has to come after your part and these method don't exist yet

Reply:
the method name now is weird now because i cannot refactor now with so many parallel branch around me

Reply:
alright, noted.